### PR TITLE
fix(docs): Fix spike protection descriptions

### PR DESCRIPTION
**Pull request by schew2381**
Requires: https://github.com/getsentry/sentry/pull/57639
1. Fix Create + Update Spike Protection body param descriptions
   - Link to List integrations page
   - Make `Required` part more visible by putting on new line
   
The params are still not in the perfect order due to an issue with getting body params from ModelSerializers. I think the body params show up in the order they're specified on in the Model itself, and not the order the fields are defined in on the ModelSerializer. The post-process hook at least puts the required params first.
   
### Create Before + After
Screenshot 2023-10-05 at 2 31 34 PM
Screenshot 2023-10-05 at 2 31 20 PM

### Update Before + After
Screenshot 2023-10-05 at 2 31 47 PM
Screenshot 2023-10-05 at 2 31 58 PM

### CODE REVIEW

File: src/sentry/apidocs/hooks.py [hunk 1]
```diff
@@ -222,6 +225,12 @@ def custom_postprocessing_hook(result: Any, generator: Any, **kwargs: Any) -> An
                     else:
                         schema = content["application/json"]["schema"]
 
+                    # If the schema is a reference, we need to resolve it
+                    if len(schema) == 1 and "$ref" in schema:
```
Comment (JoshFerge) [inline src/sentry/apidocs/hooks.py:229]: would be good to split this out and add a test

Comment (schew2381) [reply src/sentry/apidocs/hooks.py]: I moved the function into a helper. Regarding testing, I tried using the `generate_schema` function we copied over from drf-spec with the test `ExampleEndpoint`, but for some reason the `requestBody` field isn't building in the schema. It works in the drf-spec tests tho so I'm not sure what we're missing 😐

Comment (JoshFerge) [reply src/sentry/apidocs/hooks.py]: sorry, meant we should split this into a separate PR. but helper function is good too.

Comment (schew2381) [reply src/sentry/apidocs/hooks.py]: whoops ok, new PR is here: https://github.com/getsentry/sentry/pull/57639

File: src/sentry/api/serializers/rest_framework/notification_action.py [hunk 2]
```diff
@@ -236,7 +240,7 @@ def validate_slack_channel(
         if timed_out:
             raise serializers.ValidationError(
                 {
-                    "target_identifier": "Please provide a slack channel id, we encountered an error while earching for it via the channel name."
+                    "target_identifier": "Please provide a slack channel id, we encountered an error while searching for it via the channel name."
```
Comment (sentaur-athena) [inline src/sentry/api/serializers/rest_framework/notification_action.py:243]: 🤦‍♀️

**Review by sentaur-athena — APPROVED** [review]

**Pull request merged**